demo_client/src/oauth-nhsd-api/Pages/Allergies.cshtml.cs

```diff
@@ -19,48 +20,85 @@ namespace oauth_nhsd_api.Pages
         {
             _configuration = configuration;
         }
-        public async Task OnGet()
-        {
-            if (IsSessionPopulatedByApiResponse())
-            {
-                OrderedActiveList = GetListFromSessionData();
-            }
-            else
-            {
-                var ApiResponse = await GetApiResponse();
 
-                var UnorderedActiveList = CreateListFromJsonResponse(ApiResponse, "active");
+        
+         public async Task OnGet()
+           {
+                  if (IsSessionPopulatedByApiResponse())
+                    {
+                        OrderedActiveList = GetListFromSessionData();
+                       
+                    }
+                    else
+                    {
 
-                OrderedActiveList = UnorderedActiveList.OrderByDescending(listItem => listItem.AssertedDate).ToList();
-            }
+                        var ApiResponse = await GetApiResponse();
+                        if (ApiResponse == "Failed")
             
-            SetSessionDataFromList(OrderedActiveList);
-        }
-
+                                {
+                                   
+                                 
+                                    Response.Redirect("/Index");
+                                    
+                            
+                                    foreach(var cookie in HttpContext.Request.Cookies)
+                                    {
+
+                                        Response.Cookies.Delete(cookie.Key);
+                                    } 
+                                }
+                        else{
+
+                            var UnorderedActiveList = CreateListFromJsonResponse(ApiResponse, "active");
+
+                            OrderedActiveList = UnorderedActiveList.OrderByDescending(listItem => listItem.AssertedDate).ToList();
+                            
+                                    
+                            }
+                       
+                        }
+
+                    SetSessionDataFromList(OrderedActiveList);   
+                            
+                }    
+       
+        
         public async Task<string> GetApiResponse()
         {
             var tokenAccess = await HttpContext.GetTokenAsync("access_token");
             var tokenRefresh = await HttpContext.GetTokenAsync("refresh_token");
             var tokenExpiresAt = await HttpContext.GetTokenAsync("expires_at");
-
             HttpRequestMessage req = new HttpRequestMessage(System.Net.Http.HttpMethod.Get,
                 _configuration["NHSD:APIEndpoint"] + _configuration["NHSD:NhsNumber"]);
 
             req.Headers.Authorization = new AuthenticationHeaderValue("Bearer", tokenAccess);
 
             HttpResponseMessage NHSAPIresponse = await new HttpClient().SendAsync(req);
+            
+            if (NHSAPIresponse.StatusCode == HttpStatusCode.OK)
```

Comment:
This would cause all non 200 responses to be redirected.

Is this behaviour what we want?
Or should we make this more specific to handle token errors only?

Reply:
Yeah we don't want that. We should be looking for the very specific response code that corresponds to a faulty token (likely something between 400-401)